--- a/drivers/net/ethernet/broadcom/genet/bcmgenet.c
+++ b/drivers/net/ethernet/broadcom/genet/bcmgenet.c
@@ -37,7 +37,7 @@
 #include <linux/phy.h>
 #include <linux/platform_data/bcmgenet.h>
 
-#include <asm/unaligned.h>
+#include <linux/unaligned.h>
 
 #include "bcmgenet.h"
 
@@ -1323,7 +1323,7 @@
 	priv->eee.tx_lpi_enabled = tx_lpi_enabled;
 }
 
-static int bcmgenet_get_eee(struct net_device *dev, struct ethtool_eee *e)
+static int bcmgenet_get_eee(struct net_device *dev, struct ethtool_keee *e)
 {
 	struct bcmgenet_priv *priv = netdev_priv(dev);
 	struct ethtool_eee *p = &priv->eee;
@@ -1342,7 +1342,7 @@
 	return phy_ethtool_get_eee(dev->phydev, e);
 }
 
-static int bcmgenet_set_eee(struct net_device *dev, struct ethtool_eee *e)
+static int bcmgenet_set_eee(struct net_device *dev, struct ethtool_keee *e)
 {
 	struct bcmgenet_priv *priv = netdev_priv(dev);
 	struct ethtool_eee *p = &priv->eee;
@@ -3453,7 +3453,7 @@
 	bcmgenet_phy_pause_set(dev, priv->rx_pause, priv->tx_pause);
 
 	if (!eee) {
-		struct ethtool_eee eee_data;
+		struct ethtool_keee eee_data;
 
 		ret = bcmgenet_get_eee(dev, &eee_data);
 		if (ret == 0) {
@@ -4211,16 +4211,14 @@
 	return err;
 }
 
-static int bcmgenet_remove(struct platform_device *pdev)
+static void bcmgenet_remove(struct platform_device *pdev)
 {
-	struct bcmgenet_priv *priv = dev_to_priv(&pdev->dev);
+    struct bcmgenet_priv *priv = dev_to_priv(&pdev->dev);
 
-	dev_set_drvdata(&pdev->dev, NULL);
-	unregister_netdev(priv->dev);
-	bcmgenet_mii_exit(priv->dev);
-	free_netdev(priv->dev);
-
-	return 0;
+    dev_set_drvdata(&pdev->dev, NULL);
+    unregister_netdev(priv->dev);
+    bcmgenet_mii_exit(priv->dev);
+    free_netdev(priv->dev);
 }
 
 static void bcmgenet_shutdown(struct platform_device *pdev)
